rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -349,6 +349,17 @@ __declare_parameter_common(
     initial_value = &overrides_it->second;
   }
 
+  // If there is no initial value, then skip initialization
+  if (initial_value->get_type() == rclcpp::PARAMETER_NOT_SET) {
+    // Add declared parameters to storage (without a value)
+    parameter_infos[name].descriptor.name = name;
+    parameter_infos[name].descriptor.type = parameter_descriptor.type;
```

Comment:
```suggestion
    parameter_infos[name].descriptor.type = parameter_descriptor.dynamic_typing ? rclcpp::PARAMETER_NOT_SET: parameter_descriptor.type;
```

Reply:
I applied this with a written out if-else block since the ternary operator caused a compiler warning (1ea6ea4)